### PR TITLE
[Elastic Agent] Add elastic agent ID and version to events from filebeat and metricbeat.

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds the `elastic.agent.id`, `elastic.agent.version`, and `elastic.agent.snapshot` to all events published from filebeat and metricbeat inputs. This includes when filebeat/metricbeat is monitoring the running Elastic Agent itself.

## Why is it important?

So it's clear which Elastic Agent sent the events. Elastic Agent has never included this information with each event.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates #21121

### CODE REVIEW

**Comment by jen-huang** [comment]
This looks like it could be a fix for an issue I ran into when trying to match up agent logs to agent ID: https://github.com/elastic/kibana/issues/77189#issuecomment-698607209. @blakerouse do you think this PR is relevant to that?

**Comment by blakerouse** [comment]
@jen-huang Yes this is the fix you need.

**Review by ph — APPROVED** [review]
LGTM, @blakerouse are we missing heartbeat?

**Comment by blakerouse** [comment]
@ph Yes I did not add heartbeat, I didn't know if Uptime needed the same events and if the processors would work the same.

If we can confirm that heartbeat needs it I can add it in.

File: x-pack/elastic-agent/pkg/agent/application/info/agent_info.go [hunk 1]
```diff
@@ -44,3 +46,13 @@ func ForceNewAgentInfo() (*AgentInfo, error) {
 func (i *AgentInfo) AgentID() string {
 	return i.agentID
 }
+
+// Version returns the version for this Agent.
+func (*AgentInfo) Version() string {
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/application/info/agent_info.go:51]: we use release.Version and release.Snapshot throughout the code, so maybe we can unify the usage of these where it's possible in a followup

**Review by michalpristas — APPROVED** [review]
LGTM

File: x-pack/elastic-agent/pkg/agent/application/info/agent_info.go (hunk 1 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/application/info/agent_info.go:51]: Yeah I would like to. I only had to add this here for testing purposes as I needed a way to mock the version for the unit tests.

**Comment by ph** [comment]
@blakerouse I think we should also have the same for heartbeat. cc @andrewvc

**Comment by blakerouse** [comment]
@ph I don't believe that heartbeat is also using the new index strategy either. So we have a few things that need to be updated in the spec file for Heartbeat.

**Comment by ph** [comment]
@blakerouse I think heartbeat should be disabled in 7.10, @andrewvc WDYT?

**Comment by ruflin** [comment]
Each process run by Elastic Agent should be "forced" to add these fields. Do we have a good place to document this? 

Is this field also shipped by endpoint?

Mid term, these fields should not be added anymore through processors but by the processes (MB, FB, *) itself.

**Pull request merged**